### DISCUSSION TITLE
ClusterSpaceEvaluator instance created in CE Processor?

### DISCUSSION THREAD

**Discussion by kamronald** [CederGroupHub/smol#452] [Announcements]
Why does the ClusterExpansionProcessor create a new instance of ClusterSpaceEvaluator in the constructor? Isn't this redundant with the instance already contained in the ClusterSubspace?

**Comment by lbluque** [comment]
Great question! I'm trying to remember exactly why that choice was made. It may have been to be able to set the number of threads for each independently, or perhaps some pickling issues.

Though feel free to just  change the code to use the same `ClusterSpaceEvaluator`, run the unit-tests and check if there are any issues; and if not we can change it accordingly.
